duo-client/src/main/java/com/duosecurity/client/Util.java

```diff
@@ -86,4 +88,21 @@ public static CertificatePinner createPinner(String apiHost, String[] caCerts) {
 
     return pinner;
   }
+
+  /**
+   * Create hash byte array of message
+   *
+   * @param algorithm   The algorithm used to create the hash
+   * @param message     The text to create the hash
+   * @return a byte array
+   */
+  public static byte[] hash(String algorithm, String message) {
+    try{
+      MessageDigest digest = MessageDigest.getInstance(algorithm);
+      byte[] encodedhash = digest.digest(message.getBytes(StandardCharsets.UTF_8));
+      return encodedhash;
+    }catch(Exception e){
+      return new byte[0];
```

Comment:
If we're unable to hash the input, I can't see it being helpful to just use an empty array - this will pretty much guarantee an authentication error with a misleading error code?